### DISCUSSION TITLE
Charging status and Sleep Mode button state duration

### DISCUSSION THREAD

**Discussion by iRonin** [firstof9/openevse#340] [Q&A]
Hello again. 
I'd like to get the duration of the sleep state of OpenEVSE so that I don't enable the charging too frequently but I can't use the last_updated or last_changed because, I think, the values are force updated - see the screenshots where you see that last update and duration of the current state does not match.
The values and behavior are the same for `Charging Status` and `Sleep Mode` toggle.

<img width="585" alt="image" src="https://github.com/firstof9/openevse/assets/27929/c2042434-e5b1-4ec2-b78c-e5090f37677c">
<img width="554" alt="image" src="https://github.com/firstof9/openevse/assets/27929/66373be5-f898-4fb1-a6d8-4135181e6930">
<img width="555" alt="image" src="https://github.com/firstof9/openevse/assets/27929/58b9ec68-ec1b-4a8f-aa5c-500e17dc133a">

Are you using force_update on these and do you need to use it for some reason?

The easiest thing for me would be to check if `Sleep Mode` toggle is `OFF` and how long it has been in this state but this does not work. Is there an easier way than creating a SQL sensor for this?

Thanks.

**Comment by firstof9** [comment]
What do you mean by "force updated" they are updated via the websocket connection to the OpenEVSE.

> Is there an easier way than creating a SQL sensor for this?

I have no idea, that's more of a general Home Assistant question than a question about the integration.

**Comment by iRonin** [comment]
Issue similar to this one https://community.home-assistant.io/t/entity-state-last-changed-constantly-changing-even-though-the-state-didnt/436169/14

**Reply by firstof9** [reply]
There is no `force_update` property in this code.
